tests/unit/storage/metadata/mongoclient/putObject.spec.js

```diff
@@ -221,28 +221,12 @@ describe('MongoClientInterface:putObjectVerCase1', () => {
         }, false);
     });
 
-    it('should not fail when error code is 11000 and upsertedCount is 1', done => {
+    it('should fail when error code is 11000, master operation failed', done => {
```

Comment:
- `index === 0` means the "version" document failed, not master 
- should we add a test case here for the "master operation failed" case indeed (which should not fail the operation)

Reply:
The tests were updated, i also merged the two similar test suites into one, and it covers all cases, in the right file. I think we should either split the MongoClientInterface accordingly, or use a single test file though, as it doesn't follow the usuql pattern to have different names for subparts of one (big) ts file...